Stratis.Bitcoin/Notifications/TransactionReceiver.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Notifications
+{
+    /// <summary>
+    /// This class receives transaction messages from other nodes.
+    /// </summary>
+    public class TransactionReceiver : NodeBehavior
+    {
+        private readonly TransactionNotification transactionNotification;
+        private readonly TransactionNotificationProgress notifiedTransactions;
+
+        public TransactionReceiver(TransactionNotification transactionNotification, TransactionNotificationProgress notifiedTransactions)
+        {
+            this.transactionNotification = transactionNotification;
+            this.notifiedTransactions = notifiedTransactions;
+        }
+
+        protected override void AttachCore()
+        {
+            this.AttachedNode.MessageReceived += AttachedNode_MessageReceived;
+        }
+
+        protected override void DetachCore()
+        {
+            this.AttachedNode.MessageReceived -= AttachedNode_MessageReceived;
+        }
+
+        private async void AttachedNode_MessageReceived(Node node, IncomingMessage message)
+        {
+            Guard.Assert(node == this.AttachedNode); // just in case
+
+            // check the type of message received.
+            // we're only interested in Inventory and Transaction messages.
+            var invPayload = message.Message.Payload as InvPayload;
+            if (invPayload != null)
+            {
+                await this.ProcessInvAsync(node, invPayload);
```

Comment:
No need for continuation here so you can remove the async and replace await with return.
Also when using async void you need to manage exceptions